File: boot-data-mongo/src/test/java/com/example/demo/ApplicationTests.java
package com.example.demo;


import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.context.ApplicationContext;
import org.springframework.test.web.reactive.server.WebTestClient;

@SpringBootTest()
public class ApplicationTests {

    @Autowired
    ApplicationContext ctx;

    WebTestClient client;

    @Autowired
    PostController controller;

    @BeforeEach
    public void setup() {
        /**
         this.client= WebTestClient.bindToApplicationContext(ctx)
         .configureClient()
         .build();
         **/

        this.client = WebTestClient.bindToController(controller)
                .configureClient()
                .build();
    }

    @Test
    public void getAllMessagesShouldBeOk() {
        client.get().uri("/posts").exchange()
                .expectStatus().isOk();
    }

}

File: boot-kotlin-dsl/src/test/kotlin/com/example/demo/ApplicationTests.kt
package com.example.demo


import org.junit.jupiter.api.BeforeAll
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.test.context.ContextConfiguration
import org.springframework.test.web.reactive.server.WebTestClient
import org.springframework.web.reactive.function.server.RouterFunction
import org.springframework.web.reactive.function.server.ServerResponse

@SpringBootTest(
    classes = [DemoApplication::class],
    // removed since Spring Boot 3.4.0
    //properties = ["context.initializer.classes=com.example.demo.TestConfigInitializer"]
)
@ContextConfiguration(initializers = [TestConfigInitializer::class])
class ApplicationTests {

    @Autowired
    private lateinit var routerFunction: RouterFunction<ServerResponse>

    private lateinit var client: WebTestClient

    @BeforeAll
    fun setup() {
        client = WebTestClient.bindToRouterFunction(routerFunction)
            .configureClient()
            .build()
    }

    @Test
    fun `get all posts`() {
        client.get().uri("/posts")
            .exchange()
            .expectStatus().isOk
    }

}
